pkg/kubelet/volumemanager/reconciler/reconciler.go

```diff
@@ -342,10 +342,15 @@ func (rc *reconciler) reconcile() {
 			} else {
 				// Volume is attached to node, detach it
 				if rc.controllerAttachDetachEnabled || !attachedVolume.PluginIsAttachable {
-					// Kubelet not responsible for detaching or this volume has a non-attachable volume plugin,
-					// so just remove it to actualStateOfWorld without attach.
-					rc.actualStateOfWorld.MarkVolumeAsDetached(
-						attachedVolume.VolumeName, rc.nodeName)
+					// Kubelet not responsible for detaching or this volume has a non-attachable volume plugin.
+					// Mark as detached once all pending operations, such as unmounting the global mount, have finished.
+					if !rc.operationExecutor.IsOperationPending(attachedVolume.VolumeName, volumetypes.EmptyUniquePodName) {
+						rc.actualStateOfWorld.MarkVolumeAsDetached(attachedVolume.VolumeName, attachedVolume.NodeName)
+						glog.Infof("Detached volume %q (spec.Name: %q) devicePath: %q",
+							attachedVolume.VolumeName,
+							attachedVolume.VolumeSpec.Name(),
+							attachedVolume.DevicePath)
+					}
```

Comment:
I think IsOperationPending() needs to be checked before attachedVolume.GloballyMounted. Otherwise there might be a race condition as follows.
1. Checked attachedVolume.GloballyMounted is false, but there is a pending operation trying to mount
2. Pending operation finishes, and volume is mounted
3. Check whether there is a pending operation, return false, then mark volume as detached.